python/lsst/cp/pipe/ptc.py

```diff
@@ -968,16 +965,23 @@ def errFunc(p, x, y):
                 msg = (f"\nSERIOUS: Not enough data points ({len(meanVecFinal)}) compared to the number of"
                        f"parameters of the PTC model({len(parsIniPtc)}). Setting {ampName} to BAD.")
                 self.log.warn(msg)
+                lenRawTimes = len(dataset.rawExpTimes)
+                lenNonLinPars = self.config.polynomialFitDegreeNonLinearity - 1
                 dataset.badAmps.append(ampName)
                 dataset.gain[ampName] = np.nan
                 dataset.gainErr[ampName] = np.nan
                 dataset.noise[ampName] = np.nan
                 dataset.noiseErr[ampName] = np.nan
-                dataset.nonLinearity[ampName] = np.nan
-                dataset.nonLinearityError[ampName] = np.nan
-                dataset.nonLinearityResiduals[ampName] = np.nan
-                dataset.fractionalNonLinearityResiduals[ampName] = np.nan
+                dataset.nonLinearity[ampName] = [np.nan for _ in range(lenRawTimes)]
+                dataset.nonLinearityError[ampName] = [np.nan for _ in range(lenRawTimes)]
+                dataset.nonLinearityResiduals[ampName] = [np.nan for _ in range(lenRawTimes)]
+                dataset.fractionalNonLinearityResiduals[ampName] = [np.nan for _ in range(lenRawTimes)]
                 dataset.coefficientLinearizeSquared[ampName] = np.nan
+                dataset.ptcFitPars[ampName] = [np.nan for _ in range(lenRawTimes)]
+                dataset.ptcFitParsError[ampName] = [np.nan for _ in range(lenRawTimes)]
+                dataset.ptcFitReducedChiSquared[ampName] = np.nan
+                dataset.coefficientsLinearizePolynomial[ampName] = [np.nan for _ in range(lenNonLinPars)]
+                tableArray[i, :] = [np.nan for _ in range(self.config.maxAduForLookupTableLinearizer)]
```

Comment:
For all these `for _ in range(len...)` list-comps why not just do either
`[np.nan for _ in dataset.rawExpTimes]` and dispense with the initial `len()` and the `range()`, or use
`np.full_like(dataset.rawExpTimes, np.nan, dtype=float)` to explictly make a matching nan-array?

Reply:
Also, if we're not plotting these, I'm not quite sure the shape of any of these matters rather than just having a simple `nan` actually.